monarch/monarch.lua

```diff
@@ -1131,6 +1130,22 @@ function M.preload(id, options, cb)
 	return true -- return true for legacy reasons (before queue existed)
 end
 
+--- Check if a screen has missing resources, always returns false for factory
+-- @param id (string|hash) - Id of the screen to preload
+function M.has_missing_resources(id)
+	assert(id, "You must provide a screen id")
+	id = tohash(id)
+	assert(screens[id], ("There is no screen registered with id %s"):format(tostring(id)))
+
+	local screen = screens[id]
+	if screen.proxy then
+		local missing_resources = collectionproxy.missing_resources(screen.proxy)
+		if #missing_resources > 0 then
+			return true
+		end
```

Comment:
Minor thing, but I would prefer if this was changed to:

```
return #missing_resources > 0
```

Reply:
Done :)